server/task_executor.go

```diff
@@ -158,8 +161,29 @@ func runMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.
 		return true, nil, err
 	}
 
+	// For tenant mode project, we will only write back latest schema file on the last task.
+	tenantWriteBack := true
+	project, err := server.composeProjectByID(ctx, task.Database.ProjectID)
```

Comment:
BTW, the project has already been composed in the outermost scheduler

https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/server/task_scheduler.go?L116

However, this is too hard to track, so adding a compose here is safer.

I have opened an separate issue to address this
https://linear.app/bbteam/issue/BYT-181/distinguish-between-not-composed-and-composed-object

Reply:
Thanks. As a rough thought, we can have a storage utility that all objects used in business logic are all "composed".